Icebnb/Icebnb/FilterViewController.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  FilterViewController.swift
+//  Icebnb
+//
+//  Created by Robert Mogos on 06/09/2017.
+//  Copyright © 2017 Robert Mogos. All rights reserved.
+//
+
+import UIKit
+import KUIPopOver
+
+public typealias FilterBlock = (UIViewController) -> Void
+
+class FilterViewController: UITableViewController, KUIPopOverUsable {
+  
+  let filterActions: [String: FilterBlock]
+  let filters: [String]
+  
+  init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?, filters: [String: FilterBlock]) {
```

Comment:
I am curious about this naming convention `nibNameOrNil`. Isn't the "Nil" part reflected already by the fact that it is a string **optional**? Same for `nibBundleOrNil`. Just curious.

Reply:
The constructor is auto-generated by XCode. The naming is legacy from ObjC where optional was not a "thing"